dashboard/modules/resource/tests/test_resource.py

```diff
@@ -0,0 +1,244 @@
+import time
+import ray
+import requests
+import pytest
+
+from ray.dashboard.tests.conftest import *  # noqa
+from ray._private.test_utils import (
+    format_web_url,
+    wait_until_server_available,
+)
+
+
+@ray.remote(num_cpus=1, memory=1000, resources={"custom_a": 1})
+def f():
+    while True:
+        time.sleep(10)
+
+
+@ray.remote(num_cpus=1, num_gpus=1)
+def g():
+    while True:
+        time.sleep(10)
+
+
+@ray.remote(num_cpus=1, resources={"custom_b": 1})
+class Actor:
+    pass
+
+
+@pytest.mark.parametrize(
+    "ray_start_cluster", [{"include_dashboard": True}], indirect=True
+)
+def test_resource_summary_and_usage(ray_start_cluster):
+    cluster = ray_start_cluster
+    cluster.add_node(num_cpus=32, num_gpus=4, resources={"custom_a": 16})
+    cluster.add_node(num_cpus=32, resources={"custom_b": 16})
+
+    context = ray.init(address=cluster.address)
+    assert wait_until_server_available(context["webui_url"]) is True
+    webui_url = context["webui_url"]
+    webui_url = format_web_url(webui_url)
+
+    gs = [g.remote() for _ in range(4)]
+    actors = [Actor.remote() for _ in range(16)]
+    # Make sure tasks requiring special resources have been placed
+    time.sleep(1)
```

Comment:
Please remove time.sleep from test!

Reply:
Hmm, just not sure the state in the GCS would have been updated on time... don't want a flakey test...